yb-voyager/src/utils/gcs/gcsutils.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright (c) YugaByte, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package gcs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+var client *storage.Client
+
+func createClientIfNotExists() {
```

Comment:
can we move `client creation` logic to the `init()` function in this package instead of calling it multiple times from different places.

Since `init` function are called automatically on the first call to that package.
similar suggestion for s3.

Reply:
As discussed, not possible to do this because `init()` will run whenever the `utils` package is used anywhere but that shouldn't happen.